File: frontend/src/types/emergencyOrg.ts
// 应急组织类型（对应 backend/app/schemas/emergency_org.py）

/** 应急角色下的人员简要信息（后端 load_emergency_org 展开返回）。 */
export interface EmergencyMemberBrief {
  id: string;
  name?: string | null;
  phone?: string | null;
  position?: string | null;
  email?: string | null;
  org_node_id?: string | null;
}

/** 组内应急角色（总指挥/副总指挥/组长/组员…）。 */
export interface EmergencyRole {
  id?: string;
  name: string;
  duties?: string | null;
  sort_order?: number;
  /** 质检必填角色（总指挥/副总指挥），企业可自定义角色名后仍能判定。 */
  is_required?: boolean;
  member_ids: string[];
  members?: EmergencyMemberBrief[];
}

/** 应急组织单元：应急指挥部 / 各应急小组，靠 parent_id 组树。 */
export interface EmergencyUnit {
  id?: string;
  parent_id?: string | null;
  name: string;
  duties?: string | null;
  sort_order?: number;
  roles: EmergencyRole[];
}

/** 可指派成员（复用 GET /org/members/available）。 */
export interface AssignableMember {
  id: string;
  name: string;
  email: string | null;
  role: string;
  position: string | null;
  org_path: string;
}

File: backend/app/schemas/emergency_org.py
"""应急组织接口契约。字段名与前端 types/emergencyOrg.ts 一一对应。"""

from pydantic import BaseModel, Field


class EmergencyRoleIn(BaseModel):
    id: str | None = None
    name: str
    duties: str | None = None
    sort_order: int = 0
    is_required: bool = False
    member_ids: list[str] = Field(default_factory=list)


class EmergencyUnitIn(BaseModel):
    id: str | None = None
    parent_id: str | None = None
    name: str
    duties: str | None = None
    sort_order: int = 0
    roles: list[EmergencyRoleIn] = Field(default_factory=list)


class EmergencyOrgUpdate(BaseModel):
    units: list[EmergencyUnitIn] = Field(default_factory=list)


class EmergencyMemberBrief(BaseModel):
    id: str
    name: str | None = None
    phone: str | None = None
    position: str | None = None
    email: str | None = None
    org_node_id: str | None = None


class EmergencyRoleOut(EmergencyRoleIn):
    id: str
    member_ids: list[str] = Field(default_factory=list)
    members: list[EmergencyMemberBrief] = Field(default_factory=list)


class EmergencyUnitOut(EmergencyUnitIn):
    id: str
    roles: list[EmergencyRoleOut] = Field(default_factory=list)
